### PR TITLE
Refactor to remove `_.kebabCase`

### PR DESCRIPTION
**Pull request by stephenwade**
> Which issue, if any, is this issue related to?

Ref #4412.

> Is there anything in the PR that needs further explanation?

No.

### CODE REVIEW

**Review by mattxwang — COMMENTED** [review]
Generally LGTM, though I'm curious if we should add a couple of test cases since we're now providing the function ourselves. Any thoughts?

(also, I'm surprised that this is our only use of `kebabCase`!)

File: lib/utils/checkInvalidCLIOptions.js [hunk 1]
```diff
@@ -45,6 +44,22 @@ const suggest = (all, invalid) => {
...
+ */
+const kebabCase = (opt) => {
+	const matches = opt.match(/[A-Z]?[a-z]+|[A-Z]|[0-9]+/g);
+
+	if (matches) {
+		return matches.map((s) => s.toLowerCase()).join('-');
+	}
+
+	return '';
+};
```
Comment (ybiquitous) [inline lib/utils/checkInvalidCLIOptions.js:61]: [suggestion] Can we use a simpler version according to the following `kebab-case` package?

https://github.com/joakimbeng/kebab-case/blob/21cba70cd5e694470cc01402b9df829bec1cada7/index.js#L5-L9

**Comment by ybiquitous** [comment]
> Generally LGTM, though I'm curious if we should add a couple of test cases since we're now providing the function ourselves. Any thoughts?

I think the existing test cases cover, but if shortage, it would be good to add more cases to `checkInvalidCLIOptions.test.js`:

https://github.com/stylelint/stylelint/blob/403248c5586d6015127b5f98cf93442efc7c2547/lib/utils/__tests__/checkInvalidCLIOptions.test.js#L29-L39

File: lib/utils/checkInvalidCLIOptions.js (hunk 1 above)
Comment (stephenwade) [reply lib/utils/checkInvalidCLIOptions.js:61]: The linked code is only a little bit simpler, and I think my code is better at handling edge cases.

My function:
```
> kebabCase('abcDef')
'abc-def'
> kebabCase('AbcDef')
'abc-def'
> kebabCase('Option2')
'option-2'
> kebabCase('Abc123defG')
'abc-123-def-g'
```

The linked function:
```
> kebabCase('abcDef')
'abc-def'
> kebabCase('AbcDef')
'-abc-def'
> kebabCase('Option2')
'-option2'
> kebabCase('Abc123defG')
'-abc123def-g'
```

Comment (ybiquitous) [reply lib/utils/checkInvalidCLIOptions.js:61]: I see 👍🏼

**Review by ybiquitous — APPROVED** [review]
Thanks! LGTM 👍🏼

**Pull request merged**